### DISCUSSION TITLE
We need to think about how we "share data" by learning from HydroShare & Logan River Observatory

### DISCUSSION THREAD

**Discussion by joewheaton** [Riverscapes/riverscapes-tools#268] [General]
@philipbaileynar here at USU we have Jeff Horsbough, Bethany Nielson and David Tarbotton, all of whom are the architects behind [CUASHI's Hydroshare](https://www.hydroshare.org/) and the local [Logan River Observatory](https://uwrl.usu.edu//lro/index). Most of what they've mastered is how to deliver real-time and historic spot based station data in the form of time series. We should 
- get ideas from their work as we shift the Data Warehouse into primetime ;
- adopt some of their standards where appropriate;
- cross reference their data as useful;
- Get our data from our warehouse cross-referenced out of their observatory and/or Hydroshare!
 
I know that not all of it will apply, but from an outsider perspective, I want to have a better answer to the question: "hey aren't you guys all from Utah State? Are you guys collaborating together or using the same standards?"   Right now, the answer is we have been too busy to talk to each other about this shit. That is not good enough. The big difference is they have a track record and their stuff works, whereas we're getting established in the public data sharing, trusted provider space. We need to learn from them and not just reinvent on our own!

More to the point, if @MattReimer asked me what functionality I want and need for our warehouse, it is all here (minus our definitions for what riverscapes projects are). I know this begs the questions:
- Should we fork [their code and repos](https://github.com/hydroshare/) for our purposes?
- Should we collaborate and buy in to their infrastructure to deploy our spin on things within?

Short of it is, this is the best thing I can think of as a template of functionality we are going to **need** and I do mean *need*, not just *want*. 

----------
# Observatory stuff:

## Landing page for Data:

![image](https://user-images.githubusercontent.com/9723905/113905175-1cd8ff00-9790-11eb-8b13-c7eef5874d18.png)

## Data Site
![image](https://user-images.githubusercontent.com/9723905/113905696-a8eb2680-9790-11eb-92f4-f3d02e38f755.png)

--------
# Hydroshare

## Nice Dataset Page Standard

We should update some of our warehouse "project" windows or pages based on ideas from [this](https://www.hydroshare.org/resource/534b3d21ef4c499f8f89df4d8880ca20/)

![image](https://user-images.githubusercontent.com/9723905/113906833-e9976f80-9791-11eb-8ace-92e91fa493a0.png)

I see no reason why we should adopt all these concepts in the core of our Riverscapes Project definition structure. @MattReimer?

## Nice User Profile Info
I love our Google Profile stuff, but we should collect (optionally) some additional information:
![image](https://user-images.githubusercontent.com/9723905/113907169-47c45280-9792-11eb-9733-8747fa7f00af.png)

## My Resources:

![image](https://user-images.githubusercontent.com/9723905/113908060-5b23ed80-9793-11eb-9887-14960a15395e.png)

## Good Discovery of Resources

![image](https://user-images.githubusercontent.com/9723905/113908464-cc63a080-9793-11eb-905d-b003d96d86dc.png)

**Comment by MattReimer** [comment]
This looks interesting and it has significant overlap with what we're doing... to the point where I'm wondering if we need to ask ourselves some hard questions about the warehouse we're using.

Since we're so connected I wonder if we could get a lot of space (we'll need it) for cheap.
https://help.hydroshare.org/about-hydroshare/policies/quota/

Regardless, we should talk about this.

**Reply by MattReimer** [reply]
I've done some looking into it and I think we have to have a more serious discussion

Here's a dummy VBET project I uploaded into Hydroshare
https://www.hydroshare.org/resource/4bef54a5e8f94bc1952072b9f67a9e0c/

There's good API Tools
https://hs-restclient.readthedocs.io/en/latest/


Hydroshare is based on irods which might solve a lot of problems for us and take a large amount of tech-debt off our hands.
https://irods.org/

> "The Integrated Rule-Oriented Data System (iRODS) is open source data management software used by research, commercial, and governmental organizations worldwide."

https://www.youtube.com/watch?v=x4XXIhJ3kYc

**Reply by joewheaton** [reply]
@MattReimer thanks so much for this digging.  That VBET project is very surprising how much stuff you get by simply putting it in.  
I just reached out to Jeff and left a message. I also watched the irods video. Yes, we need to have a serious discussion. To me, the big questions are:
1. If this does 90% of what we're after, should we pivot some of our warehouse efforts towards something like this?
2. I don't like the non-commercial aspect of some of this.. .we can discuss a little more later. Free users and sharing is important... but even researchers are doing a piss poor job of using this.
3. Is it better to buy-in to hydroshare, or copy and deploy? 
4. How difficult is it to leverage certain pieces of this?
5. What does this not do that we need and want? Although it is easy to get excited about something new,  we have some really good stuff and I don't want to abandon that.
6. What control do we sacrifice? How much are we paying for our own control? Is it worth it?

I look forward to discussing.
